File: Android/java/r2chill/Model/Friend.java
package example.r2chill.Model;

import android.graphics.Color;

import java.util.ArrayList;
import java.util.Date;

public class Friend {
    private String ownerUsername;
    private String alias;
    private String description;
    private String textColor;
    private CurrentStatus currentStatus;
    private boolean acknowledged;

    private ArrayList<String> listOfGroupIds;

    public Friend(String userName, String name, String descript, String color,
                  CurrentStatus currentStatus, ArrayList<String> groupList) {
        this.ownerUsername = userName;
        this.alias = name;
        this.description = descript;
        this.textColor = color;
        this.currentStatus = currentStatus;
        this.acknowledged = false;
        this.listOfGroupIds = groupList;
    }

    public Friend() {
        this.ownerUsername = null;
        this.alias = null;
        this.description = null;
        this.textColor = "#FFFFFF";
        this.currentStatus = new CurrentStatus();
        this.acknowledged = false;
        this.listOfGroupIds = new ArrayList<String>();
    }

    public boolean isR2C() {
        return this.currentStatus.isR2C();
    }

    public boolean isBusy() {
        return this.currentStatus.isBusy();
    }

    public void setOwnerUsername(String newEmail) {
        ownerUsername = newEmail;
    }

    public String getOwnerUsername() {
        return ownerUsername;
    }

    public void setAlias(String newAlias) {
        alias = newAlias;
    }

    public String getAlias() {
        return alias;
    }

    public void setDescription(String newDescription) {
        description = newDescription;
    }

    public String getDescription() {
        return description;
    }

    public void setTextColor(String color) {
        this.textColor = color;
    }

    public String getTextColor() {
        return this.textColor;
    }
    public int getTextColorInt() {
        if (this.textColor.length() == 8) {
            return Color.parseColor("#" + this.textColor.substring(2));
        }
        return Color.parseColor(this.textColor);
    }

    public void setCurrentStatus(CurrentStatus updatedStatus) {
        this.currentStatus = updatedStatus;
    }

    public CurrentStatus getCurrentStatus() {
        return currentStatus;
    }

    public void setAcknowledged(boolean ack) {
        this.acknowledged = ack;
    }

    public boolean isAcknowledged() {
        return this.acknowledged;
    }

    public void setListOfGroupIds(ArrayList<String> list) {
        this.listOfGroupIds = list;
    }
    public ArrayList<String> getListOfGroupIds() {
        return this.listOfGroupIds;
    }
}

File: iOS/Chill With Me/Models/Friend.swift
//
//  Friend.swift
//  Chill With Me
//
//  Created by Phillip Hoang on 2018-09-17.
//  Copyright © 2018 IC Heart Technologies. All rights reserved.
//

import Foundation

class Friend: Codable {
    
    var ownerUsername: String?
    var alias: String?
    var description: String?
    var textColor: String
    var currentStatus: CurrentStatus
    var acknowledged: Bool
    
    var listOfGroupIds: [String]
    
    init(userName: String, name: String, descript: String, color: String, currentStatus: CurrentStatus, groupList: [String]) {
        self.ownerUsername = userName
        self.alias = name
        self.description = descript
        self.textColor = color
        self.currentStatus = currentStatus
        self.listOfGroupIds = groupList
        self.acknowledged = false
    }
    
    init() {
        self.ownerUsername = nil
        self.alias = nil
        self.description = nil
        self.textColor = "#FFFFFF"
        self.currentStatus = CurrentStatus.init()
        self.listOfGroupIds = [String]()
        self.acknowledged = false
    }
    
    func isR2C() -> Bool {
        return self.currentStatus.isR2C()
    }
    
    func isBusy() -> Bool {
        return self.currentStatus.isBusy()
    }
    
    func isBusyOrR2C() -> Bool {
        return self.currentStatus.isBusyOrR2C()
    }
}
















